pandas/core/algorithms.py

```diff
@@ -652,8 +662,13 @@ def factorize(
     # responsible only for factorization. All data coercion, sorting and boxing
     # should happen here.
 
+    if isinstance(values, ABCRangeIndex):
```

Comment:
this was moved in the PR but not on 1.1.x so may need to remove

Reply:
RangeIndex.factorize was added in #38034, so this causes a recursion error, removing.